core/server/proxy/src/main/java/alluxio/proxy/s3/ListBucketResult.java

```diff
@@ -0,0 +1,394 @@
+ */
+@JacksonXmlRootElement(localName = "ListBucketResult")
+@JsonPropertyOrder({ "Name", "Prefix", "ContinuationToken", "NextContinuationToken",
+    "KeyCount", "MaxKeys", "IsTruncated", "Contents" })
+public class ListBucketResult {
+
+  // Name of the bucket.
+  private String mName = "";
+  // Keys that begin with the indicated prefix.
+  private String mPrefix = null;
+  // ContinuationToken is included in the response if it was sent with the request.
+  private String mContinuationToken = null;
+  // Returns the number of keys included in the response. The value is always less than or equal
+  // to the mMaxKeys value.
+  private int mKeyCount = 0;
+  // The maximum number of keys returned in the response body.
+  private int mMaxKeys = S3RestUtils.S3_DEFAULT_MAX_KEYS;
+  // Specifies whether or not all of the results were returned. If the number of
+  // results exceeds that specified by MaxKeys, all of the results might not be returned.
+  private boolean mIsTruncated = false;
+  // If only partial results are returned (i.e. mIsTruncated = true), this value is set as the
+  // next continuation token. The continuation token is the full Alluxio path of the object.
+  private String mNextContinuationToken = null;
+  // A list of objects with metadata.
+  private List<Content> mContents = new ArrayList<>();
+
+  /**
+   * Creates an {@link ListBucketResult}.
+   */
+  public ListBucketResult() {}
+
+  /**
+   * Creates an {@link ListBucketResult}.
+   *
+   * @param bucketName the bucket name
+   * @param objectsList a list of {@link URIStatus}, representing the objects
+   * @param options the list bucket options
+   */
+  public ListBucketResult(
+      String bucketName, List<URIStatus> objectsList, ListBucketOptions options) {
+    mName = bucketName;
+    mPrefix = options.getPrefix();
+    mKeyCount = 0;
+    if (options.getMaxKeys() != null) {
+      mMaxKeys = Integer.parseInt(options.getMaxKeys());
+    }
+    mContents = new ArrayList<>();
+    mContinuationToken = options.getContinuationToken();
+
+    Collections.sort(objectsList, new URIStatusComparator());
+
+    int startIndex = 0;
+    if (options.getContinuationToken() != null) {
+      URIStatus tokenStatus = new URIStatus(new FileInfo().setPath(
+          mName + AlluxioURI.SEPARATOR + mContinuationToken));
+      startIndex = Collections.binarySearch(objectsList, tokenStatus, new URIStatusComparator());
+      if (startIndex < 0) {
+        startIndex = 0;
+      }
+    }
+
+    final DateFormat s3DateFormat = new SimpleDateFormat(S3RestUtils.S3_DATE_FORMAT_REGEXP);
+    for (int i = startIndex; i < objectsList.size(); i++) {
+      URIStatus status = objectsList.get(i);
+      String objectKey = status.getPath().substring(mName.length() + 1);
+      if (mKeyCount >= mMaxKeys) {
+        mIsTruncated = true;
+        mNextContinuationToken = objectKey;
+        return;
+      }
+      if (mContinuationToken != null && objectKey.compareTo(mContinuationToken) < 0) {
```

Comment:
if we returned when `startIndex < 0`, then this condition will never be satisfied since objectKey is after mContinuationToken which is enforced by the binary search